mlir/include/mlir/Dialect/MIOpen/Tuning/GridwiseGemmParams.h

```diff
@@ -363,6 +366,7 @@ class PopulateParamsBase {
       dataPerThreadCopyGemmPos2 = dataPerThreadCopyGemmVectorized;
       derived.srcVectorReadDim = 2;
     }
+    assert(derived.srcVectorReadDim != 0);
```

Comment:
Does this assertion serve only to make sure you've set `derived.srcVectorReadDim` in each branch of the `if`? If so, would a comment be called for to make sure people know why this is here?

Reply:
The value 0,1,2 is pretty bad. I’ll fix them with constant literals